app/models/user.rb

```diff
@@ -81,6 +81,29 @@ class User < ApplicationRecord
 
     filtered_records
   }
+  scope :filter_by_role, ->(role, _user = nil) { where(role:) }
+  scope :filter_by_additional_responsibilities, lambda { |additional_responsibilities, _user|
+    filtered_records = all
+    scopes = []
+
+    additional_responsibilities.each do |responsibility|
+      case responsibility
+      when "key_contact"
+        scopes << send("is_key_contact")
+      when "data_protection_officer"
+        scopes << send("is_data_protection_officer")
+      end
```

Comment:
`send` is only required when the string you pass is dynamic (its a function that works a lot like eval in other languages). since we're using a constant string we can just reference the scope directly
```suggestion
      when "key_contact"
        scopes << is_key_contact
      when "data_protection_officer"
        scopes << is_data_protection_officer
      end
```

Reply:
I've fixed this as we've now handed over